### PR TITLE
Hypershift deployment on PowerVS

### PR DESCRIPTION
**Pull request by pravin-dsilva**
Signed-off-by: Pravin D'silva <pravin.dsilva@ibm.com>

### CODE REVIEW

File: playbooks/roles/hypershift/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+- name: Clone hypershift github repo
+  git:
+    repo: "{{ hypershift_git_repo }}"
+    dest: "{{ hypershift_dir }}"
+    version:  "{{ hypershift_git_branch }}"
```
Comment (varad-ahirwadkar) [inline playbooks/roles/hypershift/tasks/main.yml:6]:
```suggestion
    version: "{{ hypershift_git_branch }}"
```

File: playbooks/roles/hypershift/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# hypershift default variables
+hypershift_install: false
+hypershift_destroy_cluster: false
```
Comment (yussufsh) [inline playbooks/roles/hypershift/defaults/main.yml:3]: Shouldn't this be `hypershift_destroy` instead?

File: playbooks/roles/hypershift/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+mgmt_cluster_kubeconfig: "{{ ansible_env.HOME }}/.kube/config"
+hosted_cluster_name: "rdr-hypershift-cluster"
+hypershift_dir: "/root/hypershift"
+hypershift_git_repo: "https://github.com/openshift/hypershift"
+hypershift_git_branch: "main"
+ibmcloud_resource_group: ""
+powervs_region: ""
+powervs_zone: ""
+vpc_region: ""
+base_domain: ""
```
Comment (yussufsh) [inline playbooks/roles/hypershift/defaults/main.yml:13]: As these are required variables there is no use defining the empty default values which may lead to failures.

File: examples/hypershift_destroy.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+mgmt_cluster_kubeconfig: "{{ ansible_env.HOME }}/.kube/config"
+hosted_cluster_name: "rdr-hypershift-cluster"
+hypershift_dir: "/root/hypershift"
+hypershift_git_repo: "https://github.com/openshift/hypershift"
+hypershift_git_branch: "main"
+ibmcloud_resource_group: ""
+powervs_region: ""
+powervs_zone: ""
+vpc_region: ""
+base_domain: ""
```
Comment (yussufsh) [inline examples/hypershift_destroy.yaml:12]: Can we show some example values here? These are required so cannot expect the user to pass an empty string.

File: examples/hypershift_install.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+powervs_zone: ""
+vpc_region: ""
+base_domain: ""
+nodepool_replicas: 2
+ocp_pull_secret_path: "~/.pullSecret"
+sys_type: "s922"
+proc_type: "dedicated"
+processors: "0.5"
+memory: "32"
+ocp_release_image: ""
```
Comment (yussufsh) [inline examples/hypershift_install.yaml:20]: Please provide an example value.

File: playbooks/roles/hypershift/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+nodepool_replicas: 2
+ocp_pull_secret_path: "~/.pullSecret"
+sys_type: "s922"
+proc_type: "dedicated"
+processors: "0.5"
+memory: "32"
+infra_id: "rdr-hypershift-infra"
+ocp_release_image: ""
```
Comment (yussufsh) [inline playbooks/roles/hypershift/defaults/main.yml:21]: Again, the default empty value will fail the create command. Better to remove it from here.

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Review by yussufsh — APPROVED** [review]
/lgtm

**Pull request merged**